### ISSUE TITLE
Can Print.js provide a third-party extension point between print.send, and the moment it finishPrint

### ISSUE THREAD

**Issue by mini-peanut** [crabbly/Print.js#189]
hi, Here I Go Again，😂i have a idea as the title says, Can print.js provide a callback like middleware that accept printDocument as a parameter and return printDocument that has been decorated，If that' all right， i can make a pull request for this

**Comment by crabbly** [comment]
Hey @mini-peanut , thank you for bringing more ideas into this project.
This sounds like something it should be done before sending the document to print.js. 
What I mean is, I can manipulate documents as much as I want, then when ready, print the document using the lib.

Is there any specific reasons why we should integrate such functionality within the library?

**Comment by mini-peanut** [comment]
wow，thank you responding me so quickly，Hm, I mean maybe sometimes，if I want a feature that it doesn't have yet or I will be more troublesome based on the current function，i can do it by myself but don't have to hack into library

**Comment by crabbly** [comment]
Can you give an example?

**Comment by mini-peanut** [comment]
such as noPrintSelector，append， prepend, manuallyCopyFormValues，deferred，etc，and just in personal view, i think it's difficult and unnecessary to contain everything

**Comment by mini-peanut** [comment]
😂

**Comment by crabbly** [comment]
Got it. Thank you for explaining. 
I still think these functions should run outside the library scope.
For example, most HTML "print" modification can be done either via a css print media query.
So, we probably won't implement any integration for now.

However, since you mentioned form values, could you please take a look at issues #186 and #187 ?
Maybe you could help with that. We need to find a way to better process the html when cloning the elements into the iframe, in order to print them properly. Right now, the `select` and `checkbox` elements are not printing properly.
Thanks again for you help.

**Comment by mini-peanut** [comment]
ok，thank you too
